test/action_view/component/base/unit_test.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class ActionView::Component::Base::UnitTest < Minitest::Test
```

Comment:
@juanmanuelramallo I added a new unit_test file to isolate the `templates` method and stub out several unique paths. I'm not sure if you want to add unit tests or relocate this test to an existing file. Tell me what you think 😄

Reply:
This is good. But I feel it would be better to maintain the current approach of testing. What do you think?

That would be about creating new components in `/test/app/components` and then use those in your test cases in `/test/action_view/component_test.rb` so that we see can assert that the output of the rendering is what we expect.

Those new components can have the path names you used in here:
- fake.user/path/to.templates/component/test_component.html+phone.erb
- /_underscore-dash./component/test_component.html+desktop.slim
- /tilda~/component/test_component.html.haml

Reply:
Sounds good. I'll rework this test and follow the existing testing patterns.

Reply:
@juanmanuelramallo I'm running into an issue with initializing the component due to Rails conventions.

I have the component and its respective view in the directory path `test/app/components/unconventional.path/_underscore-dash./tilda~/`. When I run the test for this unconventional path:
```ruby
  def test_renders_components_with_less_conventional_paths
    with_variant :phone do
      result = render_inline(UnconventionalPathComponent)

      assert_includes result.text, "Phone"
    end
  end
```
I get a NameError `uninitialized constant ActionView::ComponentTest::UnconventionalPath`. This makes sense since the component is not name spaced using the unconventional directory names. Any ideas on how to get this test to work?

Reply:
Sorry for the delay, I lost track of time. 

Maybe we have two options:
1. Use modules to namespace according to the folders (following rails convention)
2. Require the file directly in the spec_helper or in the test file

Anyways I now see that the way you tested it is great! So maybe we can make this go forward as it is, and then we can add some tests following the other approach in a follow up PR